File: 0443-string-compression/0443-string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:

        final_index = 0
        index = 0

        while index< len(chars):

            current_char = chars[index]
            count=0

            while index < len(chars) and current_char == chars[index]:
                index+=1
                count+=1
            
            chars[final_index]=current_char
            final_index+=1

            if count!=1:
                for s in str(count):
                    chars[final_index]=s
                    final_index+=1
        
        return final_index
        

File: 0443-string-compression/0443-string-compression.java
class Solution {
    public int compress(char[] chars) {
        
        int finalIndex=0;
        int index=0;
        
        while(index<chars.length) {
            
            char currentChar = chars[index];
            
            int count=0;
            
            while(index<chars.length && chars[index]==currentChar) {
                index++;
                count++;
            }
            
            chars[finalIndex++] = currentChar;
            
            if(count!=1) {
                for(char c: String.valueOf(count).toCharArray()){
                    
                    chars[finalIndex++]=c;
                    
                }
                    
            }
            
            
        }
        return finalIndex;
    }
}